src/components/DataList/DataList.jsx

```diff
@@ -1,19 +1,32 @@
 import PropTypes from "prop-types";
 
-export const DataList = ({ list }) => {
+export const DataList = ({ list, register, required }) => {
```

Comment:
I see that you didn't check what is inside the `register`. `react-hook-form` is based on uncontrolled components, that is why it has so low render score. In shortcut, it depends on `references`. After destructuring `register` you should have there `{ onChange, onBlur, name }` and `ref` on the outside. Thanks to that, while using register on the component itself, for example `<Datalist label='Datalist' options={OPTIONS} {...register('datalist')}>` in the component code you should have `{ label, options, onChange, onBlur, name } ref` in props.